logmiddleware_test.go

```diff
@@ -224,13 +254,37 @@ func logAssertions(t *testing.T, logEntry *logrus.Entry, expected ExpectedLogFie
 }
 
 func incomingRequestAssertions(t *testing.T, incomingRequestLogEntry *logrus.Entry, expected ExpectedIncomingLogFields) {
-	assert.Equal(t, incomingRequestLogEntry.Data["url"], expected.URL, "Unexpected http uri path for log in incoming request")
-	assert.Equal(t, incomingRequestLogEntry.Data["method"], expected.Method, "Unexpected http method for log in incoming request")
-	assert.Equal(t, incomingRequestLogEntry.Data["hostname"], expected.Hostname, "Unexpected hostname for log of request completed")
+	request := incomingRequestLogEntry.Data["http"].(map[string]interface{})["request"].(map[string]interface{})
+	assert.Equal(t, request["method"], expected.Method, "Unexpected http method for log in incoming request")
+	userAgent := request["userAgent"].(map[string]interface{})
```

Comment:
We could have a function to cast to `map[string]interface{}` (if we do not use structs)